pyaedt/q3d.py

```diff
@@ -368,6 +370,355 @@ def edit_sources(
+            else:
+                self.logger.error("Matrix list parameters is empty, can't export a valid matrix.")
+                return False
+
+        if r_unit is None:
+            r_unit = "ohm"
+        else:
+            if not r_unit.endswith("ohm"):
+                self.logger.error("Provide a valid unit for resistor.")
+                return False
+
+        if l_unit is None:
+            l_unit = "nH"
+        else:
+            if not l_unit.endswith("H"):
+                self.logger.error("Provide a valid unit for inductor.")
+                return False
+
+        if c_unit is None:
+            c_unit = "pF"
+        else:
+            if c_unit not in ["fF", "pF", "nF", "uF", "mF", "farad"]:
+                self.logger.error("Provide a valid unit for Capacitance.")
+                return False
+
+        if g_unit is None:
+            g_unit = "mho"
+        else:
+            if g_unit not in [
+                "fSie",
+                "pSie",
+                "nSie",
+                "uSie",
+                "mSie",
+                "Sie",
+                "kSie",
+                "megSie",
+                "mho",
+                "perohm",
+                "apV",
+            ]:
+                self.logger.error("Provide a valid unit for Conductance.")
+                return False
+
+        if freq is None:
+            freq = (
+                re.compile(r"(\d+)\s*(\w+)")
+                .match(
+                    self.modeler._odesign.GetChildObject("Analysis")
+                    .GetChildObject(setup_name)
+                    .GetPropValue("Adaptive Freq")
+                )
+                .groups()[0]
+            )
+        else:
+            if freq_unit != self.odesktop.GetDefaultUnit("Frequency") and freq_unit is not None:
+                freq = go.parse_dim_arg("{}{}".format(freq, freq_unit), self.odesktop.GetDefaultUnit("Frequency"))
+
+        if export_AC_DC_res is None:
+            export_AC_DC_res = False
+
+        if precision is None:
+            precision = 15
+        else:
+            if not isinstance(precision, int):
+                self.logger.error("Precision type must be integer.")
+                return False
+
+        if field_width is None:
+            field_width = 20
+        else:
+            if not isinstance(field_width, int):
+                self.logger.error("Field width type must be integer.")
+                return False
+
+        if use_sci_notation is None:
+            use_sci_notation = 1
+        else:
+            if use_sci_notation:
+                use_sci_notation = 1
+            else:
+                use_sci_notation = 0
+
+        if not self.modeler._is3d:
+            if length_settings is None:
+                length_settings = "Distributed"
+            else:
+                if length_settings not in ["Distributed", "Lumped"]:
+                    self.logger.error("Length setting is not valid.")
+                    return False
```

Comment:
The default value in the method signature fro `length_settings` will always be `"Distributed"`.
```suggestion
            if length_settings not in ["Distributed", "Lumped"]:
                self.logger.error("Length setting is not valid.")
                return False
```